spec/e2e/has_many_spec.rb

```diff
@@ -581,6 +581,31 @@ def false_callback(_other)
     end
   end
 
+  describe 'checking if associations are propagated to child classes' do
+    before(:each) do
+      stub_active_node_class('Thing') do
+        has_many :out, :parents, origin: :things
+      end
+
+      stub_active_node_class('OtherThing') do
+        has_many :out, :children, origin: :things
+      end
+
+      stub_active_node_class('Parent') do
+        has_one :in, :thing, type: :HAS_THINGS, model_class: :Thing, unique: true
+      end
+
+      stub_named_class('Child', Parent) do
+        has_many :in, :other_things, type: :HAS_OTHER_THINGS, model_class: :OtherThing, unique: true
+        validates :things, presence: true
+      end
+    end
+
+    it 'should return the same associations' do
+      expect(Parent.associations).to eq(Child.associations)
```

Comment:
@cheerfulstoic I think I did a mess here. It's **very** strange that this spec is passing 👎 👎  I'll take a look.